app/validators/user_role_name_validator.rb

```diff
@@ -0,0 +1,9 @@
+class UserRoleNameValidator < ActiveModel::Validator
```

Comment:
I'm curious why you pulled this out into a separate validator class? It like it's doing exactly what it was before. I'm not sure the extra indirection is that valuable.